PokeAlarm/Filters/QuestFilter.py

```diff
@@ -0,0 +1,79 @@
+# Standard Library Imports
+import operator
+# 3rd Party Imports
+# Local Imports
+from . import BaseFilter
+from PokeAlarm.Utilities import GymUtils as GymUtils
+
+class QuestFilter(BaseFilter):
+    """ Filter class for limiting which quests trigger a notification. """
+
+    def __init__(self, mgr, name, data):
+        """ Initializes base parameters for a filter. """
+        super(QuestFilter, self).__init__(mgr, 'quest', name)
+
+        # Distance
+        self.min_dist = self.evaluate_attribute(  # f.min_dist <= m.distance
+            event_attribute='distance', eval_func=operator.le,
+            limit=BaseFilter.parse_as_type(float, 'min_dist', data))
+        self.max_dist = self.evaluate_attribute(  # f.max_dist <= m.distance
+            event_attribute='distance', eval_func=operator.ge,
+            limit=BaseFilter.parse_as_type(float, 'max_dist', data))
+            
+        # Quest Description
+        self.quest_contains = self.evaluate_attribute(  # f.gn matches e.gn
+            event_attribute='quest', eval_func=GymUtils.match_regex_dict,
+            limit=BaseFilter.parse_as_set(
+                GymUtils.create_regex, 'quest_contains', data))
+        self.gym_name_excludes = self.evaluate_attribute(  # f.gn no-match e.gn
```

Comment:
I thought quests only came from pokestops?
Why does this reference gyms instead of stops?
Glad you're adding filtering by stop name though 👍

Reply:
quests do only come from stops. It only references to contains function in the gym_utils as didn't find it anywhere else, lazy I know.

I haven't added filtering by stop name?? Only quest description and reward description.

The filtering needs more work, I don't use it (not enough user submissions to bother), and just added a couple of filters last minute for those that might need it - no idea if they even work....

I did this in about 30 minutes to get quest reporting working on my server. Needs more work I know, but people kept asking me for it, so easier to PR and let people fill in the blanks.

Reply:
I should have read a little more, and I need to start drinking coffee, my bad -- `'quest_contains'` 😅 
I wasn't expecting it to work the same way as it would by stops, but it makes sense now that I think about it
We will add more filters later once we actually have it merged, so let's focus on getting this ship-shape.
Something we should consider (maybe in the future if @kvangent doesn't deem it necessary now) is moving create_regex (as well as the other regex functions for good measure) into the GenUtils file since it will be used for more than just gyms, and to eliminate my confusion
Likewise, we will probably want to add some specific names for quests, since using `gym_name_excludes` doesn't really fit the bill - maybe make something more generic? such as a `contains` and `excludes` function somewhere

Reply:
Oops, gym_name_excludes shouldn't be there lol

Reply:
Gone.